src/components/reusables/Reusable.tsx

```diff
@@ -0,0 +1,70 @@
+import React, { ChangeEvent } from 'react';
+import styles from './Reusable.module.scss';
+
+interface ButtonProps {
+  size?: string;
+  color?: string;
+  text: string;
+  onClick?: () => void;
+  type?: 'button' | 'submit' | 'reset' | undefined;
+}
+
+export const Button: React.FC<ButtonProps> = (props) => {
+  const { size, color, text, onClick, type } = props;
+  const buttonSize = styles[`size--${size || 'default'}`];
+  const buttonColor = styles[`color--${color || 'primary'}`];
+
+  return (
+    <button
+      className={`${styles.btn} ${buttonSize} ${buttonColor}`}
+      onClick={onClick}
+      type={type || 'submit'}
```

Comment:
```suggestion
      type={type || 'submit'}
      {...restProps}
```
This can allow the button to be disabled, allow many other props that you have specified